scripts/rddepman.ts

```diff
@@ -0,0 +1,211 @@
+const GITHUB_REPO = process.env.GITHUB_REPOSITORY?.split('/')[1] || 'rancher-desktop';
+
+type VersionComparison = {
+  name: string;
+  currentVersion: string | AlpineLimaISOVersion;
+  latestVersion: string | AlpineLimaISOVersion;
+};
+
+const dependencies: Dependency[] = [
+  new tools.KuberlrAndKubectl(),
+  new tools.Helm(),
+  new tools.DockerCLI(),
+  new tools.DockerBuildx(),
+  new tools.DockerCompose(),
+  new tools.DockerProvidedCredHelpers(),
+  new tools.Trivy(),
+  new tools.GuestAgent(),
+  new tools.Steve(),
+  new tools.RancherDashboard(),
+  new tools.ECRCredHelper(),
+  new LimaAndQemu(),
+  new AlpineLimaISO(),
+  new WSLDistro(),
+  new HostResolverHost(), // we only need one of HostResolverHost and HostResolverPeer
+  new MobyOpenAPISpec(),
+];
+
+function git(...args: string[]): number | null {
+  const name = 'Rancher Desktop Dependency Manager';
+  const email = 'donotuse@rancherdesktop.io';
+  const result = spawnSync('git', args, {
+    env: {
+      ...process.env,
+      GIT_AUTHOR_NAME:     name,
+      GIT_AUTHOR_EMAIL:    email,
+      GIT_COMMITTER_NAME:  name,
+      GIT_COMMITTER_EMAIL: email,
+    },
+  });
+
+  if (result.error) {
+    throw result.error;
+  }
+
+  return result.status;
+}
+
+function printable(version: string | AlpineLimaISOVersion): string {
+  return typeof version === 'string' ? version : version.isoVersion;
+}
+
+function getBranchName(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): string {
+  return `rddepman/${ name }/${ printable(currentVersion) }-to-${ printable(latestVersion) }`;
+}
+
+function getTitle(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): string {
+  return `rddepman: bump ${ name } from ${ printable(currentVersion) } to ${ printable(latestVersion) }`;
+}
+
+/**
+ * Compares the versions of two dependencies, as defined in DependencyVersions.
+ * @returns true if equal, false if not equal
+ */
+function compareVersions(version1: string | AlpineLimaISOVersion, version2: string | AlpineLimaISOVersion): boolean {
+  if (typeof version1 === 'string' && typeof version2 === 'string') {
+    return version1 === version2;
+  } else if (typeof version1 !== 'string' && typeof version2 !== 'string') {
+    return version1.isoVersion === version2.isoVersion && version1.alpineVersion === version2.alpineVersion;
+  }
+  throw new Error(`Types of version1 (${ version1 }) and version2 (${ version2 }) differ.`);
+}
+
+async function createDependencyBumpPR(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): Promise<void> {
+  const title = getTitle(name, currentVersion, latestVersion);
+  const branchName = getBranchName(name, currentVersion, latestVersion);
+
+  console.log(`Creating PR "${ title }".`);
+  await getOctokit().rest.pulls.create({
```

Comment:
This doesn't want to create PRs for me.

> Creating PR "rddepman: bump helm from 3.9.4 to 3.10.0".
RequestError [HttpError]: Validation Failed: {"resource":"PullRequest","field":"head","code":"invalid"}

Looking at my fork, the expected branch wasn't pushed.

Please note that there seems to be a bunch of branches pushed to _this_ repo with the `rddepman/` prefix; please delete them (they look stale with incorrect git authorship).

Reply:
Turns out we needed `contents: write` added to the GHA permissions. I was able to run it successfully on my fork. Hopefully it goes the same for you now.